src/SmoFacade/Server.cs

```diff
@@ -168,10 +168,34 @@ public Database Database(string dbName)
       return database == null ? null : new Database(database, this);
     }
 
+    public void Restore(SingleBackup fullBackup, string databaseName,
+      Func<int, TimeSpan> retryDurationProvider,
+      Func<string, string> fileRelocation = null)
+    {
+      var retryPolicy = Policy
+        .Handle<ExecutionFailureException>(e => e.InnerException != null
+                                                && e.InnerException is SqlException
+                                                && e.InnerException.Message
+                                                  .Contains("The process cannot access the file because it is being used by another process"))
+        .WaitAndRetry(10, retryDurationProvider);

```

Comment:
`retryPolicy` can be placed into a private variable and then re-used by both restore functions.
Maybe rename to `_restoreRetryPolicy`?

Reply:
looks like it needs `retryDurationProvider` (maybe a private utility method then)?